packages/core/lib/debug/printer.js

```diff
@@ -171,6 +171,17 @@ class DebugPrinter {
     const colorizedLines = splitLines(colorizedSource);
 
     this.config.logger.log("");
+    // printout a warning message and display the end of source code when the instruction's
+    // byte-offset to the start of the range in the source code is past the end of source code
+    const instruction = this.session.view(solidity.current.instruction);
+    if (instruction.start > source.length - 1) {
+      this.config.logger.log(
+        `${colors.bold(
+          "Warning:"
+        )} The end of source code is displayed since the instruction's source range is past the end of source code.`
```

Comment:
I think the wording here could be tightened up a bit; perhaps just something like "Warning: Past end of source, displaying end"?  This is a computer printout aimed at fellow programmers, we don't need to be too elaborate here. :)